mne_bids/utils.py

```diff
@@ -28,7 +28,72 @@
 from mne.time_frequency import psd_array_welch
 
 from mne_bids.config import BIDS_PATH_ENTITIES, reader
-from mne_bids.tsv_handler import _to_tsv, _tsv_to_str
+from mne_bids.tsv_handler import (_to_tsv, _tsv_to_str,
+                                  _from_tsv, _drop)
+
+
+def delete_scan(bids_basename, bids_root, verbose=True):
+    """Safely delete a scan inside bids root.
+
+    Deleting a scan that conforms to the bids-validator, will
+    delete both the row in the `scans.tsv` file, and also
+    the corresponding sidecar files and the data file itself.
```

Comment:
what about `participants.tsv`?

Reply:
Deleting scan shouldn't delete participants.tsv entry unless it's the absolute last scan for that participant. With that being said tho... that should be incorporated. 

Do we want `delete_participant` as part of the `write` API as well then?